packages/ui-calendar/src/Calendar/Day/index.tsx

```diff
@@ -45,7 +45,7 @@ import generateComponentTheme from './theme'
 type Props = {
   makeStyles?: (...args: any[]) => any
   styles?: any
-  date: any // TODO: I18nPropTypes.iso8601
```

Comment:
Not really.
It is possible theoretically but it would involve huge effort on our side to implement it properly. We would need a custom ISO8601 type, then we would need to create some sort of factory helper function which would return this `ISO8601` type from input strings, something like this:
```ts
function toISO8601(input: string): ISO8601 {
...logic to parse and validate the input string
}
```
Then, we would need to expose this function to our users, and they would need to use it in order to have proper type safety. IMHO it is not worth it, since we already have a PropTypes runtime validator for this exact case:
https://github.com/instructure/instructure-ui/blob/1dbc3eda5dd24fbc0704aa63437e44e6d7fae3ae/packages/ui-i18n/src/I18nPropTypes.ts#L28-L64

And we are using it in this component already:
https://github.com/instructure/instructure-ui/blob/1dbc3eda5dd24fbc0704aa63437e44e6d7fae3ae/packages/ui-calendar/src/Calendar/Day/index.tsx#L80-L84

Reply:
Well, in this case we probably have to keep this proptype checker :)

Reply:
or with template literals... but this might be too complex for them.. https://github.com/freewind-demos/typescript-typing-string-literal-template-iso-date-demo

Reply:
Yes, the problem is that the ISO8601 is too broad as a standard so I think we can't really express it with the literal types.